core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/StoreCallbacksImpl.scala

```diff
@@ -581,6 +584,36 @@ object StoreCallbacksImpl extends StoreCallbacks with Logging with Serializable
       } else userId
     } else userId
   }
+
+  private lazy val samplerClass = {
+    val samplerClassName = "org.apache.spark.sql.execution.StratifiedSampler$"
+    spark.util.Utils.classForName(samplerClassName)
+  }
+
+  private lazy val removeSamplerMethod: Method = {
+    // Iterating over the method list to find the `removeSampler` method because when loaded
+    // using reflection, the type of the second parameter of this method is `boolean`
+    // (and not `scala.Boolean`). Unable to figure out a way to load this method using
+    // `java.lang.Class.getMethod` method.
+    samplerClass.getMethods.filter(m => {
+      val paramTypes = m.getParameterTypes
+      if (m.getName.equals("removeSampler") && m.getParameterCount == 2 &&
+          paramTypes(0).equals(classOf[String]) && paramTypes(1).getName.equals("boolean")) {
+        true
+      } else false
+    }).head
+  }
+
```

Comment:
samplerClass.getMethod("removeSampler", classOf[String], classOf[java.lang.Boolean]) works fine

Reply:
Somehow it's not working on my machine. I already tried earlier and tried again. Failing with: ```java.lang.NoSuchMethodException: org.apache.spark.sql.execution.StratifiedSampler$.removeSampler(java.lang.String, java.lang.Boolean)```

Could this be a JVM version/vendor specific issue?  Java vesion on my machine: `openjdk version "1.8.0_222"`

Reply:
Just tried and the following worked: `getMethod("removeSampler", classOf[String], classOf[Boolean])`

Also don't really need to go to the singleton class since bytecode of the main class also has static method, so:

```scala
    val samplerClassName = "org.apache.spark.sql.execution.StratifiedSampler"

    ...
    samplerClass.getMethod("removeSampler", classOf[String], classOf[Boolean])
    ...
    removeSamplerMethod.invoke(null, sampleTableName, Boolean.box(false))
```